### PR TITLE
Awsrequests replace

### PR DESCRIPTION
**Pull request by byron70**
- Switches out the old awsrequests library for the new signing library that does sigv4 with boto.  
- Updated Travis CI to support up to python 3.7.
- Removed python 3.3, as setup tools does not support this version, at least pypi indicates this is the case. And 3.3 builds were failing.

### CODE REVIEW

**Comment by byron70** [comment]
Reopening this one so we could just use cd-build-tools to do a release.

File: seshypy/version.py [hunk 1]
```diff
@@ -1,2 +1 @@
-"""Package version will be written by setup.py."""
-VERSION = 'undefined'
+VERSION = '0.6.dev'
```
Comment (byron70) [inline seshypy/version.py]: This needs backing out.

Comment (byron70) [reply seshypy/version.py]: reverted

File: seshypy/base_session.py [hunk 2]
```diff
@@ -87,7 +91,8 @@ def __enable_method_cache(self, method_name):
             try:
                 return not inspect.isfunction(method.cache_info)
             except AttributeError:
-                setattr(self, method_name, safe_ttl_cache(ttl=self.cache_ttl)(method))
+                setattr(self, method_name, safe_ttl_cache(
+                    ttl=self.cache_ttl)(method))
```
Comment (theherk) [inline seshypy/base_session.py]: I think your indentation is not PEP-8, but I could be wrong. In either case, not a blocker.

File: seshypy/session.py [hunk 3]
```diff
@@ -175,7 +186,8 @@ def new_and_call(*args, **kwargs):
                 service = method.__self__.__class__.__name__.lower()
                 client = self.__client(service)
             except UnknownServiceError:
-                service = method.__self__.__class__.__name__.split('.')[0].lower()
+                service = method.__self__.__class__.__name__.split('.')[
+                    0].lower()
```
Comment (theherk) [inline seshypy/session.py]: Is a tool doing this. Definitely making it less readable.

Comment (byron70) [reply seshypy/session.py]: yeah that one is horrid. VS Code. I can fix that.

File: seshypy/base_session.py (hunk 2 above)
Comment (byron70) [reply seshypy/base_session.py]: I checked, it's actually a PEP-8 linter in VSCode doing this and the other henky one with the list index.

Comment (byron70) [reply seshypy/base_session.py]: fixed this

**Comment by theherk** [comment]
All looks good to me. I'm not so sure about the need for tox, but totally okay with it. @defond0, I'll make sure you have the perms you need.

**Pull request merged**